packages/editor/src/store/reducer.js

```diff
@@ -148,6 +148,9 @@ export const editor = flow( [
 	} )( ( state = { value: [] }, action ) => {
 		switch ( action.type ) {
 			case 'RESET_EDITOR_BLOCKS':
+				if ( action.blocks === state.value ) {
```

Comment:
When does this happen

Reply:
It happens if we do `onInput( blocks )` and then `onChange( blocks )` with the same blocks. For example, when you type in a RichText and then stop typing for 1 second.

Reply:
> It happens if we do `onInput( blocks )` and then `onChange( blocks )` with the same blocks. For example, when you type in a RichText and then stop typing for 1 second.

It also seems at odds maybe with the implementation of `withHistory`, where the higher-order reducer [will abort creating a history entry](https://github.com/WordPress/gutenberg/blob/cd82d07a70c5485396427f7196e78ae5c2d3b284/packages/editor/src/utils/with-history/index.js#L111-L113) if there is no change in state, when by `onChange` we're saying "always create a history entry".